src/components/AnonymousChat/child/ChatItemMyTextV2.tsx

```diff
@@ -115,55 +51,140 @@ const ChatItemMyTextV2 = ({
   message = '',
   status = ChatStatus.PENDING,
   avatar,
-  chatType
+  chatType,
+  messageType,
+  data
 }: ChatItemMyTextProps) => {
-  const messageRef = React.useRef<Text>(null);
+  const {
+    setReplyPreview,
+    onContextMenuPressed,
+    pulseAnimation,
+    animatedBubbleStyle,
+    animatedPulseStyle
+  } = useMessageHook();
+  const scrollContext = React.useContext(ScrollContext);
+
+  const swipeableRef = React.useRef<Swipeable>(null);
+  const [isNewLine, setIsNewLine] = React.useState(true);
+
+  React.useEffect(() => {
+    if (scrollContext?.selectedMessageId === data?.id) {
+      pulseAnimation.value = withSequence(
+        withDelay(300, withTiming(1.1, {duration: 200})),
+        withTiming(1, {duration: 200})
+      );
+      scrollContext?.setSelectedMessageId(null);
+    }
+  }, [scrollContext?.selectedMessageId]);
+
+  const isReply = messageType === MESSAGE_TYPE_REPLY;
+  const isReplyPrompt = messageType === MESSAGE_TYPE_REPLY_PROMPT;
+  const isShowUserInfo = !isContinuous || isReplyPrompt || isReply;
+
+  const onTextLayout = (event: NativeSyntheticEvent<TextLayoutEventData>) => {
+    const {lines} = event.nativeEvent;
+    const lastLine = lines[lines.length - 1];
+    const lastLineWidth = lastLine?.width;
+    const isCalculatedNewLine = targetLastLineWidth - lastLineWidth < 24;
+    setIsNewLine(isCalculatedNewLine);
+  };
 
   const renderIcon = React.useCallback(() => {
     if (status === ChatStatus.PENDING)
       return (
         <View style={styles.icon}>
-          <IconChatClockGrey width={12} height={12} />
+          <IconChatClockGrey color={colors.silver} width={12} height={12} />
         </View>
       );
 
     return (
       <View style={styles.icon}>
-        <IconChatCheckMark />
+        <IconChatCheckMark color={colors.silver} />
       </View>
     );
-  }, []);
+  }, [status]);
+
+  const contextMenuActions: ContextMenuAction[] = [
+    {title: CONTEXT_MENU_REPLY, systemIcon: 'arrow.turn.up.left'}
+  ];
+
+  if (messageType !== MESSAGE_TYPE_DELETED) {
+    contextMenuActions.push({title: CONTEXT_MENU_COPY, systemIcon: 'square.on.square'});
+    contextMenuActions.push({title: CONTEXT_MENU_DELETE, systemIcon: 'trash', destructive: true});
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section where you're defining the context menu actions is clear and straightforward. However, it would be more efficient to move this array outside of the component or wrap it in a `useMemo` hook to prevent unnecessary re-renders.

```diff
+  const contextMenuActions: ContextMenuAction[] = useMemo(() => [
    {title: CONTEXT_MENU_REPLY, systemIcon: 'arrow.turn.up.left'},
    ...(messageType !== MESSAGE_TYPE_DELETED ? [
      {title: CONTEXT_MENU_COPY, systemIcon: 'square.on.square'},
      {title: CONTEXT_MENU_DELETE, systemIcon: 'trash', destructive: true}
    ] : [])
+  ], [messageType]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->